### DISCUSSION TITLE
tutorial for creating model and ... some biomechanical example

### DISCUSSION THREAD

**Discussion by ANaaim** [stack-of-tasks/pinocchio#2484] [Q&A] [answered]
Hello,

I am exploring tools for conducting multibody dynamics calculations in our lab and am interested in what can be achieved with Pinocchio. I would like to know if there are any tutorials available for creating a model. I might not have looked thoroughly enough, but I couldn’t find any examples on how to create a “model” to be used in Pinocchio.

My basic usage would involve processing data from our gait lab efficiently and eventually moving towards real-time musculoskeletal modeling. Again, I might not have looked thoroughly enough, but I couldn’t find any examples on how to create a “model” to be used in Pinocchio.

Your repository indicates that your tools are used in biomechanical applications, but I couldn’t find any examples relevant to what I want to do. Are there any available?

Currently, I would like to be able to:

Define all solids.
Define all joints. As I understand it, in Pinocchio, you define the joints and the solids are not defined (just their properties attached to the joint object), so this should be technically done before defining the solids.
Add some external forces.
Perform inverse kinematics and inverse dynamics to get moments at the different joints to begin with.
Our basic files are C3D files with marker coordinates (markers attached to body segments) and some force data extracted from force platforms.

I found some videos on YouTube from a winter school, but it was impossible to read what was written on the whiteboard, so I gave up.

Thanks in advance,,

**Comment by jcarpent** [comment, marked as answer]
You can find an example for creating your own model here: https://github.com/stack-of-tasks/pinocchio/blob/42306ed023b301aafef91e2e76cb070c5e9c3f7d/examples/simulation-inverted-pendulum.py
Otherwise, I invite you to start with the tutorials https://github.com/stack-of-tasks/pinocchio/?tab=readme-ov-file#tutorials

**Reply by ANaaim** [reply]
Thanks for your answer, i will have a look at it.
